hack/verify-godeps.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+branch="${1:-master}"
+# notice this uses ... to find the first shared ancestor
+if ! git diff origin/"${branch}"...HEAD | grep 'Godeps/' > /dev/null; then
+  exit 0
+fi
+
+# create a nice clean place to put our new godeps
+_tmpdir="$(mktemp -d -t gopath.XXXXXX)"
+function cleanup {
+  echo "Removing ${_tmpdir}"
+  rm -rf "${_tmpdir}"
+}
+trap cleanup EXIT
+
+# build the godep tool
+export GOPATH="${_tmpdir}"
+go get -u github.com/tools/godep 2>/dev/null
+go install github.com/tools/godep 2>/dev/null
+GODEP="${_tmpdir}/bin/godep"
+
+# fill out that nice clean place with the kube godeps
+echo "Starting to download all kubernetes godeps. This takes a while"
```

Comment:
Do we really want to do this?  My experience with 'godep restore' is that it can take 10 minutes - I don't want to do that on every travis and 'git commit' ...